lib/fluent/plugin/sd_static.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/plugin/service_discovery'
+
+module Fluent
+  module Plugin
+    class SdStatic < ServiceDiscovery
+      Plugin.register_sd('static', self)
+
+      LISTEN_PORT = 24224
+
+      desc 'The IP address or host name of the server.'
+      config_param :host, :string
+      desc 'The name of the server. Used for logging and certificate verification in TLS transport (when host is address).'
+      config_param :name, :string, default: nil
+      desc 'The port number of the host.'
+      config_param :port, :integer, default: LISTEN_PORT
+      desc 'The shared key per server.'
+      config_param :shared_key, :string, default: nil, secret: true
+      desc 'The username for authentication.'
+      config_param :username, :string, default: ''
+      desc 'The password for authentication.'
+      config_param :password, :string, default: '', secret: true
+      desc 'Marks a node as the standby node for an Active-Standby model between Fluentd nodes.'
+      config_param :standby, :bool, default: false
+      desc 'The load balancing weight.'
+      config_param :weight, :integer, default: 60
+
+      def configure(conf)
+        super
+
+        @services << ServiceDiscovery::Service.new(:static, @host, @port, @name, @weight, @standby, @username, @password, @shared_key)
```

Comment:
Does this plugin support only one server unlike out_forward?

Reply:
Good catch.. changed to be able to handle multiple services.
https://github.com/fluent/fluentd/pull/2541/commits/be0475893869bda7a463ed1e731a3e259ad203d5